File: cash.py
from cs50 import get_float

# ask for change owed and verify if its not a negative number, otherwise prompt again
while True:
    cents = get_float('Change: ')
    if cents > 0:
        break

# convert the change in dollars to cent for easier management
cents = round(cents * 100)

# set the counter of coins to 0
count = 0

# QUARTERS: verify if the cents are greater or equal to 25, if it is reduce 25 and add 1 to count, otherwise break the loop
while True:
    if cents >= 25:
        cents = cents - 25
        count += 1
    else:
        break

# DIMES: verify if the cents are greater or equal to 10, if it is reduce 10 and add 1 to count, otherwise break the loop
while True:
    if cents >= 10:
        cents = cents - 10
        count += 1
    else:
        break

# NICKELS: verify if the cents are greater or equal to 5, if it is reduce 5 and add 1 to count, otherwise break the loop
while True:
    if cents >= 5:
        cents = cents - 5
        count += 1
    else:
        break
# PENNIES: verify if the cents are greater or equal to 1, if it is reduce 1 and add 1 to count, otherwise break the loop
while True:
    if cents >= 1:
        cents = cents - 1
        count += 1
    else:
        break

# Print the count that represent the number of coins owed
print(f'{count}')

# the else statement its not necesary, but logical correct

File: cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

int get_cents(void)
{
    // check that number is positive and get how many cents
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);
    return cents;
}

int calculate_quarters(int cents)
{
    // define int quarters = 0 to get how many quarters do the cents have until cents is less than 25, return the number of quarters
    int quarters = 0;
    while (cents >= 25)
    {
        cents = cents - 25;
        quarters++;
    }

    return quarters;
}

int calculate_dimes(int cents)
{
    // define int dimes = 0 to get how many dimes do the cents have until cent is less than 10, return the number of dimes
    int dimes = 0;
    while (cents >= 10)
    {
        cents = cents - 10;
        dimes++;
    }
    return dimes;
}

int calculate_nickels(int cents)
{
    // define int nickles = 0 to get how many nickles do the cents have until cents is less than 5, return the number of nickels
    int nickels = 0;
    while (cents >= 5)
    {
        cents = cents - 5;
        nickels++;
    }
    return nickels;
}

int calculate_pennies(int cents)
{
    // define int pennies = 0 to get how many pennies do the cents have until cents is less than 1, return the number of pennies
    int pennies = 0;
    while (cents >= 1)
    {
        cents = cents - 1;
        pennies++;
    }
    return pennies;
}